hoptimator-util/src/main/java/com/linkedin/hoptimator/util/planner/ScriptImplementor.java

```diff
@@ -245,38 +254,93 @@ public void implement(SqlWriter w) {
    * */
   class InsertImplementor implements ScriptImplementor {
     private final String name;
+    private final String schema;
     private final RelNode relNode;
+    private final ImmutableList<Pair<Integer, String>> targetFields;
 
-    public InsertImplementor(String name, RelNode relNode) {
+    public InsertImplementor(String name, String schema, RelNode relNode,
+        ImmutableList<Pair<Integer, String>> targetFields) {
       this.name = name;
+      this.schema = schema;
       this.relNode = relNode;
+      this.targetFields = targetFields;
     }
 
     @Override
     public void implement(SqlWriter w) {
       w.keyword("INSERT INTO");
       (new IdentifierImplementor(name)).implement(w);
       SqlWriter.Frame frame1 = w.startList("(", ")");
-      RelNode project = dropNullFields(relNode);
+      RelNode project = dropFields(schema, relNode, targetFields);
       (new ColumnListImplementor(project.getRowType())).implement(w);
       w.endList(frame1);
       (new QueryImplementor(project)).implement(w);
       w.literal(";");
     }
 
-    private static RelNode dropNullFields(RelNode relNode) {
+    // Drops NULL fields
+    // Drops additional fields by schema
+    private static RelNode dropFields(String schema, RelNode relNode,
+        ImmutableList<Pair<Integer, String>> targetFields) {
       List<Integer> cols = new ArrayList<>();
       int i = 0;
+      Set<String> targetFieldNames = targetFields.stream().map(x -> x.right).collect(Collectors.toSet());
       for (RelDataTypeField field : relNode.getRowType().getFieldList()) {
+        // TODO: Need a better way to dynamically modify the script implementer based on the schema
+        if (schema.startsWith("VENICE")
```

Comment:
Don't like this. The schema should be opaque. I don't understand why Venice is special here? Can we just always drop fields that are not in `targetFieldNames`? Or maybe only drop if `targetFieldNames` isn't null?

Reply:
Yea good point, I don't like this either. Let me see what the behavior is like if I leave out fields for Kafka